pkg/datasources/dynamic_tables_acceptance_test.go

```diff
@@ -0,0 +1,100 @@
+			"table_name": config.StringVariable(tableName),
+		}
+	}
+	variableSet1 := m()
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: testAccCheckDynamicTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: config.TestStepDirectory(),
+				ConfigVariables: variableSet1,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "like.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "like.0.pattern", name),
+					resource.TestCheckResourceAttr(dataSourceName, "in.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "in.0.database", acc.TestDatabaseName),
+					resource.TestCheckResourceAttr(dataSourceName, "starts_with", name),
+					resource.TestCheckResourceAttr(dataSourceName, "limit.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "limit.0.rows", "1"),
+
+					// computed attributes
+					resource.TestCheckResourceAttr(dataSourceName, "records.#", "1"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.created_on"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.database_name"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.schema_name"),
+					// unused by Snowflake API at this time (always empty)
+					// resource.TestCheckResourceAttrSet(dataSourceName, "records.0.cluster_by"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.rows"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.bytes"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.owner"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.target_lag"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.refresh_mode"),
+					// unused by Snowflake API at this time (always empty)
+					// resource.TestCheckResourceAttrSet(dataSourceName, "records.0.refresh_mode_reason"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.warehouse"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.comment"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.text"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.automatic_clustering"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.scheduling_state"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.last_suspended_on"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.is_clone"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.is_replica"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "records.0.data_timestamp"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDynamicTableDestroy(s *terraform.State) error {
```

Comment:
this is the more general solution, in case there were multiple such resources declared in the configuration. also if you did it the way you suggest, then there would need to be a wrapper function because the interface would not be the same as what is expected for a checkDestroy function